src/SheetsRegistry.js

```diff
@@ -15,7 +15,21 @@ export default class SheetsRegistry {
    * @api public
    */
   add(sheet) {
-    this.registry.push(sheet)
+    const {registry} = this
+
+    if (registry.length && sheet.options.index) {
+      for (let i = 0; i < registry.length; i++) {
+        const registration = registry[i]
+        if (
+          !registration.options.index ||
+          registration.options.index < sheet.options.index
+        ) continue
+        registry.splice(i, 0, sheet)
```

Comment:
What about the case where 2 sheets have the same index, I think the order of insertions should be the same as they come, right now, you would insert sheets as they arrive before the previous one, if I understand correctly from reading ...